src/API/FileController.ts

```diff
@@ -46,67 +46,67 @@ export interface Arguments {
   readonly configuration: FileControllerConfiguration
 }
 
-export class FileController {
-  private readonly logger: Pino.Logger
-  private readonly ipfsArchiveUrlPrefix: string
-  private readonly ipfsUrl: string
-  private readonly fileDao: FileDAO.FileDAO
-
-  constructor({
-    dependencies: {
-      fileDao,
-      logger,
-    },
-    configuration,
-  }: Arguments) {
-    this.fileDao = fileDao
-    this.logger = childWithFileName(logger, __filename)
-    this.ipfsArchiveUrlPrefix = configuration.ipfsArchiveUrlPrefix
-    this.ipfsUrl = configuration.ipfsUrl
-  }
+export interface FileController {
+  readonly addFiles: (fileStreams: ReadonlyArray<fs.ReadStream>) => Promise<{}>
```

Comment:
kind of a weird return type, it should be an array of objects, but it wouldn't let me use that. Not sure of the exact cause yet, my best guess at the moment is it has something to do with either asyncPipe or the Promise.all.bind or both.

Reply:
Looks like its actually the log functions typings causing this.